extension/runtime/src/main/java/org/iris_events/runtime/BasicPropertiesProvider.java

```diff
@@ -115,6 +115,9 @@ private AMQP.BasicProperties buildAmqpBasicPropertiesWithCustomHeaders(final AMQ
             headers.remove(ROUTER);
             headers.remove(USER_ID);
             headers.put(SESSION_ID, sessionId);
+        } else if (routingDetails.getCorrelationId() != null) {
```

Comment:
This must be applied even if session or userId are present. Otherwise the signature of the send method with options is misleading. 

I would allow this even if original message arrived through the router. Another possibility is to add new header (e.g. `x-tracking-id` or sth similar.

@ctomc your thoughts?

Reply:
Changed it to how you suggested. Done in a separate commit, can be reverted.